src/map/status.cpp

```diff
@@ -1064,6 +1064,12 @@ int status_damage(struct block_list *src,struct block_list *target,int64 dhp, in
 		unit_stop_walking( target, 1 );
 	}
 
+	if (status->hp == 0 &&
+		((skill_id == AL_INCAGI || skill_id == AL_BLESSING ||
+		  skill_id == CASH_BLESSING || skill_id == CASH_INCAGI ||
+		  skill_id == MER_INCAGI || skill_id == MER_BLESSING) && sc && sc->data[SC_CHANGEUNDEAD]))
+		status->hp = 1; // Can't be killed
```

Comment:
On official the whole skill does nothing when you are at 1 HP. In this version it doesn't kill you but still shows the skill animation and the damage.

So basically instead of calling skill_attack you should just do "break;" in that case rather than adding this code.